bound/bound.go

```diff
@@ -4,7 +4,37 @@
 
 package bound
 
+import "math"
+
 // Bound represents [Min, Max] bounds.
 type Bound struct {
 	Min, Max float64
 }
+
+// Intersection returns a intersected bound between bounds.
```

Comment:
Intersection returns a Bound that is the intersection of the input bounds. If the intersection is empty or if the input length is zero, then the NaN Bound will be returned.